packages/webapp/src/_app/eos/secret-publisher.ts

```diff
@@ -0,0 +1,239 @@
+import { PrivateKey, PublicKey } from "eosjs/dist/eosjs-jssig";
+import { generateKeyPair } from "eosjs/dist/eosjs-key-conversions";
+import { KeyType } from "eosjs/dist/eosjs-numeric";
+
+export const publishSecretToChain = async (
+    message: string,
+    publisherAccount: string,
+    recipientAccounts: string[],
+    info?: string
+) => {
+    console.info("encrypting message", message);
+    const [publisherKey, ...recipientKeys] = await fetchRecipientKeys([
+        publisherAccount,
+        ...recipientAccounts,
+    ]);
+    console.info(publisherKey, recipientKeys);
+
+    // TODO: check publisherkey matches the current one in browser
+
+    const transientKeyPair = generateKeyPair(KeyType.k1, { secureEnv: true });
+    console.info(transientKeyPair);
+
+    const keks = await ecdhRecipientsKeyEncriptionKeys(
+        [publisherKey, ...recipientKeys],
+        transientKeyPair.privateKey,
+        info
+    );
+    console.info(keks);
+
+    const sessionKey = await generateRandomSessionKey();
+    console.info("session key", sessionKey);
+
+    const encryptedSessionKeys = await encryptSessionKeys(sessionKey, keks);
+    console.info(encryptedSessionKeys);
+
+    const encryptedMessage = await encryptMessage(sessionKey, message);
+
+    // TODO: prepare publish trx to eos chain with all the prepared material
+    // eden.publishelection(
+    //     publisher = logged_in_user
+    //     round_no = '2',
+    //     encrypted_link,
+    //   // for each recipient
+    //     [(encrypted_session_keyN, recipient_keyN, transient_ecc_pubkey)]
+    //   )
+    console.info({
+        encryptedSessionKeys,
+        publisherKey,
+        recipientKeys,
+        transientKeyPair,
+        encryptedMessage,
+    });
+
+    // TODO: remove this test
+    console.info("testing decryption...");
+    await decryptPublishedMessage(
+        encryptedMessage,
+        publisherKey,
+        transientKeyPair.publicKey.toString(),
+        encryptedSessionKeys[0],
+        info
+    );
+
+    return {};
+};
+
+export const decryptPublishedMessage = async (
+    encryptedMessage: Uint8Array,
+    recipientPublicKey: string,
+    transientPublicKey: string,
+    encryptedSessionKey: Uint8Array,
+    info?: string
+) => {
+    const recipientPrivateKey = retrieveRecipientPrivateKey(recipientPublicKey);
+    const ecdhSecret = deriveEcdhSecret(
+        recipientPrivateKey,
+        PublicKey.fromString(transientPublicKey)
+    );
+    console.info("ecdh secret is ", ecdhSecret);
```

Comment:
This is fine for debugging, but in production, cryptographic secrets should never be logged.

Reply:
I plan to remove all the console prints